neo/rawio/spikeglxrawio.py

```diff
@@ -0,0 +1,315 @@
+            key = (info['seg_index'], info['stream_name'])
+            self.signals_info_dict[key] = info
+
+            # create memmap
+            data = np.memmap(info['bin_file'], dtype='int16', mode='r',
+                        shape=(info['sample_length'], info['num_chan']), offset=0, order='C')
+            self._memmaps[key] = data
+
+        # create channel header
+        self._global_channel_to_stream = {}
+        self._global_channel_to_local_channel = []
+        self._channel_location = {}
+        sig_channels = []
+        global_chan = 0
+        signal_annotations = []
+        for stream_name in stream_names:
+            # take first segment
+            info = self.signals_info_dict[0, stream_name]
+
+            group_id = stream_names.index(info['stream_name'])
+
+            # add channels to global list
+            for local_chan in range(info['num_chan']):
+                self._global_channel_to_stream[global_chan] = info['stream_name']
+                self._global_channel_to_local_channel.append(local_chan)
+                chan_name = info['channel_names'][local_chan]
+                sig_channels.append((chan_name, global_chan, info['sampling_rate'], 'int16',
+                                    info['units'], info['channel_gains'][local_chan],
+                                    info['channel_offsets'][local_chan], group_id))
+
+                # annotation
+                ann = {}
+                ann['stream'] = info['stream_name']
+                signal_annotations.append(ann)
+
+                # channel location
+                if 'channel_location' in info:
+                    self._channel_location[info['seg_index'], info['device']] = \
+                                                                                                    info['channel_location']
+
+                # the channel id is a global counter and so equivalent to channel_index
+                # this is bad : this should be changed by an id base on a str
+                global_chan += 1
+
+        sig_channels = np.array(sig_channels, dtype=_signal_channel_dtype)
+        self._global_channel_to_local_channel = np.array(
+                                    self._global_channel_to_local_channel, dtype='int64')
+
+        # No events
+        event_channels = []
+        event_channels = np.array(event_channels, dtype=_event_channel_dtype)
+
+        # No spikes
+        unit_channels = []
+        unit_channels = np.array(unit_channels, dtype=_unit_channel_dtype)
+
+        # deal with nb_segment and t_start/t_stop per segment
+        self._t_starts = {seg_index: 0. for seg_index in range(nb_segment)}
+        self._t_stops = {seg_index: 0. for seg_index in range(nb_segment)}
+        for seg_index in range(nb_segment):
+            for stream_name in stream_names:
+                info = self.signals_info_dict[seg_index, stream_name]
+                t_stop = info['sample_length'] / info['sampling_rate']
```

Comment:
This is exactly the weakness of the Io at the moment.
I ask help to spikeglx team.
They don't have many time to help.
We can dix this later because this could block the PR for a long long time.
For time alignement between stream I think they use pulse on the very last channel on each stream and then they certainly do a time regresion to ajust t_start and also sampling_rate.
Too complicated for now.

Reply:
Here's the new issue: https://github.com/NeuralEnsemble/python-neo/issues/923